core/src/main/java/jenkins/I18n.java

```diff
@@ -93,7 +93,16 @@ public HttpResponse doResourceBundle(StaplerRequest request) {
         String language = request.getParameter("language");
         String country = request.getParameter("country");
         String variant = request.getParameter("variant");
-
+        // https://www.w3.org/International/questions/qa-lang-priorities
+        // in case we have regions/countries in the language query parameter
+        if (country == null && language.length()>2) {
```

Comment:
I think you meant `language.length() >= 5`, right? See the following call to `substring` as it assumes the string is length 5.